Assets/Scripts/Player.cs

```diff
@@ -40,14 +42,43 @@ private void FixedUpdate()
             Math.Abs(moveDelta.y * Time.deltaTime), collisionMasks);
         if (raycasthit.collider == null)
         {
-            transform.Translate(0, moveDelta.y * Time.deltaTime, 0);
+            transform.Translate(0, moveDelta.y * Time.deltaTime * playerSpeed, 0);
         }
 
         raycasthit = Physics2D.BoxCast(transform.position, boxCollider.size, 0, new Vector2(moveDelta.x, 0),
             Math.Abs(moveDelta.x * Time.deltaTime), collisionMasks);
         if (raycasthit.collider == null)
         {
-            transform.Translate(moveDelta.x * Time.deltaTime, 0, 0);
+            transform.Translate(moveDelta.x * Time.deltaTime *playerSpeed, 0, 0);
+        }
+
+        _shootTimer -= Time.deltaTime;
+    }
+
+
+    public void Shoot(Vector3 mousePos)
```

Comment:
Ich denke wir sollten das ganze schiessen in ein eigenes skript nehmen.
War ein schlechter start von mir das player skript player zu nennen. Denke nicht jedes Objekt im Game hat ein eigenes Skript, sondern jedes Attribut eines Objekts.. z.b. di möglichkeit zu schiessen, die möglichkeit etwas vom boden aufzuheben, aber auch das Gesundheitsattribut, ist ein eigenes Skript... so können wir für jedes Objekt mehrere Skripte hinzufügen, und diese Definieren zusammen wie sich das Objekt verhält.

Reply:
Fair; will be done